test/powershell/Modules/Microsoft.PowerShell.Utility/Format-Table.Tests.ps1

```diff
@@ -465,4 +465,172 @@ er
             $ps.HadErrors | Should -BeFalse
             $output.Replace("`r","").Replace(" ",".").Replace("`n","``") | Should -BeExactly $expectedTable.Replace("`r","").Replace(" ",".").Replace("`n","``")
         }
-}
+
+        It "Format-Table should correctly render rows: <variation>" -TestCases @(
+            @{ view = "Default"; widths = 4,7,5; variation = "narrow values"; values = [PSCustomObject]@{First=1;Second=2;Third=3}; wrap = $false; expectedTable = @"
+
+Long*Header2*Heade
+Long*********r3
+Head
+er
+----*-------*-----
+1**********2***3
+
+
+
+"@ },
+@{ view = "Default"; widths = 4,7,5; variation = "narrow values with wrap"; values = [PSCustomObject]@{First=1;Second=2;Third=3}; wrap = $true; expectedTable = @"
+
+Long*Header2*Heade
+Long*********r3
+Head
+er
+----*-------*-----
+1**********2*3
+
+
+
+"@ },
+@{ view = "Default"; widths = 4,7,5; variation = "wide values"; values = [PSCustomObject]@{First="12345";Second="12345678";Third="123456"}; wrap = $false; expectedTable = @"
+
+Long*Header2*Heade
+Long*********r3
+Head
+er
+----*-------*-----
+1...*...5678*12...
+
+
+
+"@ },
+            @{ view = "One"; widths = 3,1,1; variation = "wide values, 1 column"; values = [PSCustomObject]@{First="12345";Second="12345678";Third="123456"}; wrap = $false; expectedTable = @"
+
+Lon
+gLo
+ngH
+ead
+er
+---
+123
+
+
+
+"@ },
+@{ view = "Default"; widths = 4,7,5; variation = "wide values with wrap"; values = [PSCustomObject]@{First="12345";Second="12345678";Third="123456"}; wrap = $true; expectedTable = @"
+
+Long*Header2*Heade
+Long*********r3
+Head
+er
+----*-------*-----
+1234*1234567*12345
+5**********8*6
```

Comment:
Do we need a test case where not all the values wrap?  Maybe have 1st and 3rd wrap, and the another where only 1st wraps.

Reply:
I agree.
Added.